apps/emqx/src/emqx_router.erl

```diff
@@ -361,17 +363,12 @@ has_route_tab_entry(Topic, Dest) ->
     [] =/= ets:match(?ROUTE_TAB, #route{topic = Topic, dest = Dest}).
 
 cleanup_routes_v1(Node) ->
-    Patterns = [
-        #route{_ = '_', dest = Node},
-        #route{_ = '_', dest = {'_', Node}}
-    ],
-    mria:transaction(?ROUTE_SHARD, fun() ->
-        [
-            mnesia:delete_object(?ROUTE_TAB, Route, write)
-         || Pat <- Patterns,
-            Route <- mnesia:match_object(?ROUTE_TAB, Pat, write)
-        ]
-    end).
+    lists:foreach(
+        fun(Pattern) ->
+            {atomic, ok} = mria:match_delete(?ROUTE_TAB, #route{_ = '_', dest = Pattern})
```

Comment:
Note: it's two separate transactions now (`mnesia:match_delete/2` is a transaction that can't be nested).